tck/features/expressions/pattern/Pattern2.feature

```diff
@@ -28,5 +28,211 @@
 
 #encoding: utf-8
 
-Feature: Pattern2 - Number of Pattern Matches
+Feature: Pattern2 - Pattern Comprehension
 
+  Scenario: [1] Returning a pattern comprehension
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A)
+      CREATE (a)-[:T]->(:B),
+             (a)-[:T]->(:C)
+      """
+    When executing query:
+      """
+      MATCH (n)
+      RETURN [p = (n)-->() | p] AS ps
+      """
+    Then the result should be, in any order:
+      | ps                                     |
+      | [<(:A)-[:T]->(:C)>, <(:A)-[:T]->(:B)>] |
+      | []                                     |
+      | []                                     |
+    And no side effects
+
+  Scenario: [2] Returning a pattern comprehension with label predicate
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A), (b:B), (c:C), (d:D)
+      CREATE (a)-[:T]->(b),
+             (a)-[:T]->(c),
+             (a)-[:T]->(d)
+      """
+    When executing query:
+      """
+      MATCH (n:A)
+      RETURN [p = (n)-->(:B) | p]
+      """
+    Then the result should be, in any order:
+      | [p = (n)-->(:B) \| p] |
+      | [<(:A)-[:T]->(:B)>]   |
```

Comment:
Since the result column looks rather weird here, I would suggest aliasing it to something
```suggestion
      RETURN [p = (n)-->(:B) | p] AS aToB
      """
    Then the result should be, in any order:
      | aToB                |
      | [<(:A)-[:T]->(:B)>] |
```